[2023-10-04 22:11:14,586] INFO [GroupMetadataManager brokerId=1] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2023-10-04 22:18:33,324] INFO [ReplicaFetcher replicaId=1, leaderId=2, fetcherId=0] Error sending fetch request (sessionId=734214820, epoch=3245) to node 2: {}. (org.apache.kafka.clients.FetchSessionHandler)
java.io.IOException: Connection to 2 was disconnected before the response was read
	at org.apache.kafka.clients.NetworkClientUtils.sendAndReceive(NetworkClientUtils.java:100)
	at kafka.server.ReplicaFetcherBlockingSend.sendRequest(ReplicaFetcherBlockingSend.scala:108)
	at kafka.server.ReplicaFetcherThread.fetchFromLeader(ReplicaFetcherThread.scala:206)
	at kafka.server.AbstractFetcherThread.processFetchRequest(AbstractFetcherThread.scala:300)
	at kafka.server.AbstractFetcherThread.$anonfun$maybeFetch$3(AbstractFetcherThread.scala:135)
	at kafka.server.AbstractFetcherThread.maybeFetch(AbstractFetcherThread.scala:134)
	at kafka.server.AbstractFetcherThread.doWork(AbstractFetcherThread.scala:117)
	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:96)
[2023-10-04 22:18:33,326] WARN [ReplicaFetcher replicaId=1, leaderId=2, fetcherId=0] Error in response for fetch request (type=FetchRequest, replicaId=1, maxWait=500, minBytes=1, maxBytes=10485760, fetchData={}, isolationLevel=READ_UNCOMMITTED, toForget=, metadata=(sessionId=734214820, epoch=3245), rackId=) (kafka.server.ReplicaFetcherThread)
java.io.IOException: Connection to 2 was disconnected before the response was read
	at org.apache.kafka.clients.NetworkClientUtils.sendAndReceive(NetworkClientUtils.java:100)
	at kafka.server.ReplicaFetcherBlockingSend.sendRequest(ReplicaFetcherBlockingSend.scala:108)
	at kafka.server.ReplicaFetcherThread.fetchFromLeader(ReplicaFetcherThread.scala:206)
	at kafka.server.AbstractFetcherThread.processFetchRequest(AbstractFetcherThread.scala:300)
	at kafka.server.AbstractFetcherThread.$anonfun$maybeFetch$3(AbstractFetcherThread.scala:135)
	at kafka.server.AbstractFetcherThread.maybeFetch(AbstractFetcherThread.scala:134)
	at kafka.server.AbstractFetcherThread.doWork(AbstractFetcherThread.scala:117)
	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:96)
[2023-10-04 22:18:33,602] ERROR Error while reading checkpoint file /tmp/kafka-logs-1/cleaner-offset-checkpoint (kafka.server.LogDirFailureChannel)
java.nio.file.NoSuchFileException: /tmp/kafka-logs-1/cleaner-offset-checkpoint
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:371)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:422)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
	at java.base/java.nio.file.Files.newInputStream(Files.java:156)
	at java.base/java.nio.file.Files.newBufferedReader(Files.java:2839)
	at java.base/java.nio.file.Files.newBufferedReader(Files.java:2871)
	at kafka.server.checkpoints.CheckpointFile.liftedTree2$1(CheckpointFile.scala:87)
	at kafka.server.checkpoints.CheckpointFile.read(CheckpointFile.scala:86)
	at kafka.server.checkpoints.OffsetCheckpointFile.read(OffsetCheckpointFile.scala:61)
	at kafka.log.LogCleanerManager.$anonfun$allCleanerCheckpoints$2(LogCleanerManager.scala:134)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:583)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:597)
	at scala.collection.mutable.ListBuffer.addAll(ListBuffer.scala:118)
	at scala.collection.mutable.ListBuffer$.from(ListBuffer.scala:38)
	at scala.collection.immutable.List$.from(List.scala:617)
	at scala.collection.immutable.List$.from(List.scala:611)
	at scala.collection.IterableFactory$Delegate.from(Factory.scala:288)
	at scala.collection.immutable.Iterable$.from(Iterable.scala:35)
	at scala.collection.immutable.Iterable$.from(Iterable.scala:32)
	at scala.collection.IterableFactory$Delegate.from(Factory.scala:288)
	at scala.collection.IterableOps.flatMap(Iterable.scala:674)
	at scala.collection.IterableOps.flatMap$(Iterable.scala:674)
	at scala.collection.AbstractIterable.flatMap(Iterable.scala:921)
	at kafka.log.LogCleanerManager.$anonfun$allCleanerCheckpoints$1(LogCleanerManager.scala:132)
	at kafka.log.LogCleanerManager.allCleanerCheckpoints(LogCleanerManager.scala:140)
	at kafka.log.LogCleanerManager.$anonfun$grabFilthiestCompactedLog$1(LogCleanerManager.scala:171)
	at kafka.log.LogCleanerManager.grabFilthiestCompactedLog(LogCleanerManager.scala:168)
	at kafka.log.LogCleaner$CleanerThread.cleanFilthiestLog(LogCleaner.scala:327)
	at kafka.log.LogCleaner$CleanerThread.tryCleanFilthiestLog(LogCleaner.scala:314)
	at kafka.log.LogCleaner$CleanerThread.doWork(LogCleaner.scala:303)
	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:96)
[2023-10-04 22:18:33,609] WARN [ReplicaManager broker=1] Stopping serving replicas in dir /tmp/kafka-logs-1 (kafka.server.ReplicaManager)
[2023-10-04 22:18:33,619] INFO [ReplicaFetcherManager on broker 1] Removed fetcher for partitions Set(topic02-2, topic01-0, topic02-1, topic02-0) (kafka.server.ReplicaFetcherManager)
[2023-10-04 22:18:33,619] INFO [ReplicaAlterLogDirsManager on broker 1] Removed fetcher for partitions Set(topic02-2, topic01-0, topic02-1, topic02-0) (kafka.server.ReplicaAlterLogDirsManager)
[2023-10-04 22:18:33,625] WARN [ReplicaManager broker=1] Broker 1 stopped fetcher for partitions topic02-2,topic01-0,topic02-1,topic02-0 and stopped moving logs for partitions  because they are in the failed log directory /tmp/kafka-logs-1. (kafka.server.ReplicaManager)
[2023-10-04 22:18:33,626] WARN Stopping serving logs in dir /tmp/kafka-logs-1 (kafka.log.LogManager)
[2023-10-04 22:18:33,630] ERROR Shutdown broker because all log dirs in /tmp/kafka-logs-1 have failed (kafka.log.LogManager)
